pyccel/codegen/codegen.py

```diff
@@ -147,7 +147,6 @@ def set_printer(self, **settings):
         # instantiate codePrinter
         code_printer = printer_registry[language]
         errors = Errors()
-        errors.set_parser_stage('codegen')
```

Comment:
As mentioned in the PR description `Errors.parser_scope` was superceded a while ago by `pyccel.utilities.stage.PyccelStage`.

`pyccel.utilities.stage.PyccelStage` stores the current stage of processing and is used in many places in the code including in the AST. `Errors.parser_scope` stores the Pyccel stage internally to the `Errors` class so that it can be used to indicate the stage in the error message. This is unnecessary duplication that was missed when `pyccel.utilities.stage.PyccelStage` was first implemented. Furthermore the stages for the errors were syntax/semantic/codegen. There was not wrapper stage. It seemed to make more sense to remove the duplication rather than add the wrapper stage to `Errors.parser_scope`